server/api.go

```diff
@@ -28,16 +29,43 @@ func NewAPI(p *Plugin) *API {
 	router.HandleFunc("/iframe/mattermostTab", api.iFrame).Methods(http.MethodGet)
 	router.HandleFunc("/iframe/authenticate", api.authenticate).Methods(http.MethodGet)
 	router.HandleFunc("/iframe/notification_preview", api.iframeNotificationPreview).Methods(http.MethodGet)
-	router.HandleFunc("/iframe-manifest", api.appManifest).Methods(http.MethodGet)
+	router.HandleFunc("/iframe-manifest", api.adminRequired(api.appManifest)).Methods(http.MethodGet)
 	router.HandleFunc("/csp-report", api.cspReport).Methods(http.MethodPost)
 
+	// Icon upload and retrieval endpoints
+	router.HandleFunc("/icons/upload", api.adminRequired(api.uploadIcon)).Methods(http.MethodPost)
+	router.HandleFunc("/icons/{iconType}", api.adminRequired(api.getIcon)).Methods(http.MethodGet)
```

Comment:
Should this endpoints be prepared with the appID as a parameter to be prepared in case we support multiple apps in the future?